### DISCUSSION TITLE
Dynamic button changes when variable status changes?

### DISCUSSION THREAD

**Discussion by tbjohn** [mpaperno/MSFSTouchPortalPlugin#55] [Q&A]
With MSFS2020 (thanks Max) I am trying to find out how to change a button icon when the sim vaiable changes. For example I have a button to change to one with a green light and 'HDG IS ON' when pressed, referring to the staus of the A/P heading hold. When pressed again it turns to a red light and the obvious message. Sometimes the autopilot will disconnect a button and I would like the relevant icon to change when it does so, e.g. VS when the desired altitude is reached etc.. I've tried everything _I_ can think of but then...
Can anyone help please?

**Comment by mpaperno** [comment]
Hi, and welcome  :)

Check out the ["starter page"](https://github.com/mpaperno/MSFSTouchPortalPlugin/wiki/Pages-Buttons-and-Graphics#starter-page-1-13-x-8-layout) from the wiki. It's full of examples like that, for instance all the AP toggles, the lights, and de-ice switches.

Perhaps obviously, you'll need the correct data from the sim to properly reflect the switch position. If the switch status (on/off) simulator variable isn't already in the list of Touch Portal States this plugin provides by default, then you'll need to find what variable in the sim reflects the switch status and read up on https://github.com/mpaperno/MSFSTouchPortalPlugin/wiki/Using-Custom-States-and-Simulator-Variables.

HTH,
-Max

**Reply by tbjohn** [reply]
Hello Max and thank you for your  swift reply.
Unfortunately I have read and tried the instructions many times, you may remember my thread #53 a couple of months ago when I was trying to make a button toggle FLC in a C172 G1000 autopilot. Firstly there was no States Editor page in my Assets tab. I downloaded it from the graphics and buttons section and followed the instructions, got the blue events message about 197 items registered, saved the .ini file etc. etc.. but the state didn't appear in the plugin, tried several times with no luck.
I just looked at you message and I don't even know what I'm looking for in the SDK. I've tried searching for 'annunciator', 'verify', 'lights' etc.. I did try another one (AUTOPILOT NAV1 LOCK) and that worked but I don't know if that was there before anyway.
I tried the 4 step cheap and cheerful version that you  kindly sent me but couldn't get that to work either, I followed the steps correctly and my screens looked the same as your instructions, saved the .ini file in Settings/settings Plugins, msfs2020.
So in the end I resorted to using the virtual keystroke command of TP's. I then tried some 'If button state changes...' which works with a green light button icon and then a red one but if the button status changes as a result of the simulator ( e.g. altitude attained) on it's own that Is no good !
So I'm lost and at the point of folding up a paper aeroplane.
But thanks for trying, it's appreciated.
Trevor.





<body>


Parameters | Universal Var Member | Description
-- | -- | --
AUTOPILOT_ACTIVE | b (BOOL) | Toggles the autopilot master on/off.
AUTOPILOT_AIRSPEED_ACQUISITION | b (BOOL) | Currently not used within the simulation.
AUTOPILOT_AIRSPEED_HOLD | b (BOOL) | Returns whether airspeed hold is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_AIRSPEED_HOLD_CURRENT | b (BOOL) | Currently not used within the simulation.
AUTOPILOT_AIRSPEED_HOLD_VAR | n (FLOAT64) | Returns the target holding airspeed (in Knots) for the autopilot.
AUTOPILOT_AIRSPEED_MAX_CALCULATED | n (FLOAT64) | Can be set to limit the maximum airspeed of the autopilot to the given kcas value.
AUTOPILOT_AIRSPEED_MIN_CALCULATED | n (FLOAT64) | Can be set to limit the minimum airspeed of the autopilot to the given kcas value.
AUTOPILOT_ALTITUDE_ARM | b (BOOL) | Returns whether the autopilot is in altitude arm mode or not.
AUTOPILOT_ALTITUDE_LOCK | b (BOOL) | Toggles the autopilot altitude lock on or off.
AUTOPILOT_ALTITUDE_MANUALLY_TUNABLE | b (BOOL) | Returns whether the autopilot hold altitude is manually tuneable (1, TRUE) or not (0, FALSE).
AUTOPILOT_ALTITUDE_SLOT_INDEX | n (FLOAT64) | Index of the slot that the autopilot will use for the altitude reference. Note that there are 3 slot indices (1, 2, 3) that you can set/get normally, however you can also target slot index 0. Writing to slot 0 will overwrite all other slots with the slot 0 value, and by default the autopilot will follow slot 0 if you have not selected any slot index.See alt_mode_slot_index for more information.
AUTOPILOT_APPROACH_ACTIVE | b (BOOL) | When TRUE (1), the autopilot is currently flying the approach Flight Plan (the last legs), and when FALSE (0) it's not.
AUTOPILOT_APPROACH_ARM | b (BOOL) | Returns TRUE (1) when the autopilot is active on the approach, once it reaches the adequate condition (in most cases, once it reaches the second-last waypoint of the flightplan).
AUTOPILOT_APPROACH_CAPTURED | b (BOOL) | Returns TRUE (1) when the lateral NAV mode is engaged and the angular deviation with the current tuned navigation frequency is less than 5°.
AUTOPILOT_APROACH_HOLD | b (BOOL) | Returns whether approach mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_ATTITUDE_HOLD | b (BOOL) | Returns whether attitude hold mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_AUTO_THROTTLE_ARM | b (BOOL) | Returns whether the autopilot auto-throttle is armed (1, TRUE) or not (0, FALSE).
AUTOPILOT_AVIONICS_MANAGED | b (BOOL) | Returns whether the autopilot has active managed avionics (1, TRUE) or not (0, FALSE).
AUTOPILOT_BACKCOURSE_HOLD | b (BOOL) | Returns whether the autopilot back course mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_BANK_HOLD_REF | n (FLOAT64) | The current bank-hold bank reference.Note that if you set this, the next frame the value will be overwritten by the engine, so you may need to write to this every game frame to ensure it maintains the required value.
AUTOPILOT_BANK_MODE | b (BOOL) | Returns whether the autopilot bank hold mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_CRUISE_SPEED | b (BOOL) | Currently not used within the simulation.
AUTOPILOT_DISENGAGED | b (BOOL) | Returns whether the autopilot has been disengaged (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_ACTIVE | b (BOOL) | Returns whether the autopilot is receiving a signal from the runway beacon and is following the slope to reach the ground (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_ARM | b (BOOL) | Returns whether the autopilot is currently active on the glideslope (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_HOLD | b (BOOL) | Returns whether the autopilot glidslope hold is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_LOCK | b (BOOL) | Returns whether the autopilot heading lock is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_LOCK_DIR | n (FLOAT64) | Specifies / Returns the locked in heading for the autopilot.
AUTOPILOT_HEADING_MANUALLY_TUNABLE | b (BOOL) | Returns whether the autopilot heading lock is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_SLOT_INDEX | n (FLOAT64) | Index of the slot that the autopilot will use for the heading reference. Note that there are 3 slot indices (1, 2, 3) that you can set/get normally, however you can also target slot index 0. Writing to slot 0 will overwrite all other slots with the slot 0 value, and by default the autopilot will follow slot 0 if you have not selected any slot index.
AUTOPILOT_MACH_HOLD | b (BOOL) | Returns whether the autopilot mach hold is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_MACH_HOLD_VAR | n (FLOAT64) | Returns the target holding mach airspeed for the autopilot.
AUTOPILOT_MANAGED_INDEX | n (FLOAT64) | Currently not used within the simulation.
AUTOPILOT_MANAGED_SPEED_IN_MACH | b (BOOL) | Returns whether the autopilot managed speed is in mach (1, TRUE) or not (0, FALSE).
AUTOPILOT_MANAGED_THROTTLE_ACTIVE | b (BOOL) | Returns whether the autopilot managed throttle is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_MAX_BANK | n (FLOAT64) | Returns the maximum banking angle for the autopilot, in radians.
AUTOPILOT_MAX_BANK_ID | n (FLOAT64) | Returns the index of the current maximum bank setting of the autopilot.
AUTOPILOT_MAX_SPEED | b (BOOL) | Currently not used within the simulation.
AUTOPILOT_NAV1_LOCK


</body>
</html>Parameters	Universal Var Member	Description
AUTOPILOT_ACTIVE	b (BOOL)	Toggles the autopilot master on/off.
AUTOPILOT_AIRSPEED_ACQUISITION	b (BOOL)	Currently not used within the simulation.
AUTOPILOT_AIRSPEED_HOLD	b (BOOL)	Returns whether airspeed hold is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_AIRSPEED_HOLD_CURRENT	b (BOOL)	Currently not used within the simulation.
AUTOPILOT_AIRSPEED_HOLD_VAR	n (FLOAT64)	Returns the target holding airspeed (in [Knot](https://docs.flightsimulator.com/html/Programming_Tools/Event_IDs/#)s) for the autopilot.
AUTOPILOT_AIRSPEED_MAX_CALCULATED	n (FLOAT64)	Can be set to limit the maximum airspeed of the autopilot to the given [kcas](https://docs.flightsimulator.com/html/Programming_Tools/Event_IDs/#) value.
AUTOPILOT_AIRSPEED_MIN_CALCULATED	n (FLOAT64)	Can be set to limit the minimum airspeed of the autopilot to the given [kcas](https://docs.flightsimulator.com/html/Programming_Tools/Event_IDs/#) value.
AUTOPILOT_ALTITUDE_ARM	b (BOOL)	Returns whether the autopilot is in altitude arm mode or not.
AUTOPILOT_ALTITUDE_LOCK	b (BOOL)	Toggles the autopilot altitude lock on or off.
AUTOPILOT_ALTITUDE_MANUALLY_TUNABLE	b (BOOL)	Returns whether the autopilot hold altitude is manually tuneable (1, TRUE) or not (0, FALSE). 
AUTOPILOT_ALTITUDE_SLOT_INDEX	n (FLOAT64)	
Index of the slot that the autopilot will use for the altitude reference. Note that there are 3 slot indices (1, 2, 3) that you can set/get normally, however you can also target slot index 0. Writing to slot 0 will overwrite all other slots with the slot 0 value, and by default the autopilot will follow slot 0 if you have not selected any slot index.

See [alt_mode_slot_index](https://docs.flightsimulator.com/html/Content_Configuration/SimObjects/Aircraft_SimO/systems_cfg.htm#alt_mode_slot_index) for more information.

AUTOPILOT_APPROACH_ACTIVE	b (BOOL)	When TRUE (1), the autopilot is currently flying the approach Flight Plan (the last legs), and when FALSE (0) it's not.
AUTOPILOT_APPROACH_ARM	b (BOOL)	Returns TRUE (1) when the autopilot is active on the approach, once it reaches the adequate condition (in most cases, once it reaches the second-last waypoint of the flightplan).
AUTOPILOT_APPROACH_CAPTURED	b (BOOL)	Returns TRUE (1) when the lateral NAV mode is engaged and the angular deviation with the current tuned navigation frequency is less than 5°.
AUTOPILOT_APROACH_HOLD	b (BOOL)	Returns whether approach mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_ATTITUDE_HOLD	b (BOOL)	Returns whether attitude hold mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_AUTO_THROTTLE_ARM	b (BOOL)	Returns whether the autopilot auto-throttle is armed (1, TRUE) or not (0, FALSE).
AUTOPILOT_AVIONICS_MANAGED	b (BOOL)	Returns whether the autopilot has active managed avionics (1, TRUE) or not (0, FALSE).
AUTOPILOT_BACKCOURSE_HOLD	b (BOOL)	Returns whether the autopilot back course mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_BANK_HOLD_REF	n (FLOAT64)	
The current bank-hold bank reference.

Note that if you set this, the next frame the value will be overwritten by the engine, so you may need to write to this every game frame to ensure it maintains the required value.

AUTOPILOT_BANK_MODE	b (BOOL)	Returns whether the autopilot bank hold mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_CRUISE_SPEED	b (BOOL)	Currently not used within the simulation.
AUTOPILOT_DISENGAGED	b (BOOL)	Returns whether the autopilot has been disengaged (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_ACTIVE	b (BOOL)	Returns whether the autopilot is receiving a signal from the runway beacon and is following the slope to reach the ground (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_ARM	b (BOOL)	Returns whether the autopilot is currently active on the glideslope (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_HOLD	b (BOOL)	Returns whether the autopilot glidslope hold is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_LOCK	b (BOOL)	Returns whether the autopilot heading lock is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_LOCK_DIR	n (FLOAT64)	Specifies / Returns the locked in heading for the autopilot. 
AUTOPILOT_HEADING_MANUALLY_TUNABLE	b (BOOL)	Returns whether the autopilot heading lock is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_SLOT_INDEX	n (FLOAT64)	Index of the slot that the autopilot will use for the heading reference. Note that there are 3 slot indices (1, 2, 3) that you can set/get normally, however you can also target slot index 0. Writing to slot 0 will overwrite all other slots with the slot 0 value, and by default the autopilot will follow slot 0 if you have not selected any slot index.
AUTOPILOT_MACH_HOLD	b (BOOL)	Returns whether the autopilot mach hold is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_MACH_HOLD_VAR	n (FLOAT64)	Returns the target holding mach airspeed for the autopilot.
AUTOPILOT_MANAGED_INDEX	n (FLOAT64)	Currently not used within the simulation.
AUTOPILOT_MANAGED_SPEED_IN_MACH	b (BOOL)	Returns whether the autopilot managed speed is in mach (1, TRUE) or not (0, FALSE).
AUTOPILOT_MANAGED_THROTTLE_ACTIVE	b (BOOL)	Returns whether the autopilot managed throttle is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_MAX_BANK	n (FLOAT64)	Returns the maximum banking angle for the autopilot, in radians.
AUTOPILOT_MAX_BANK_ID	n (FLOAT64)	Returns the index of the current maximum bank setting of the autopilot.
AUTOPILOT_MAX_SPEED	b (BOOL)	Currently not used within the simulation.
AUTOPILOT_NAV1_LOCK
<body>


Parameters | Universal Var Member | Description
-- | -- | --
AUTOPILOT_ACTIVE | b (BOOL) | Toggles the autopilot master on/off.
AUTOPILOT_AIRSPEED_ACQUISITION | b (BOOL) | Currently not used within the simulation.
AUTOPILOT_AIRSPEED_HOLD | b (BOOL) | Returns whether airspeed hold is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_AIRSPEED_HOLD_CURRENT | b (BOOL) | Currently not used within the simulation.
AUTOPILOT_AIRSPEED_HOLD_VAR | n (FLOAT64) | Returns the target holding airspeed (in Knots) for the autopilot.
AUTOPILOT_AIRSPEED_MAX_CALCULATED | n (FLOAT64) | Can be set to limit the maximum airspeed of the autopilot to the given kcas value.
AUTOPILOT_AIRSPEED_MIN_CALCULATED | n (FLOAT64) | Can be set to limit the minimum airspeed of the autopilot to the given kcas value.
AUTOPILOT_ALTITUDE_ARM | b (BOOL) | Returns whether the autopilot is in altitude arm mode or not.
AUTOPILOT_ALTITUDE_LOCK | b (BOOL) | Toggles the autopilot altitude lock on or off.
AUTOPILOT_ALTITUDE_MANUALLY_TUNABLE | b (BOOL) | Returns whether the autopilot hold altitude is manually tuneable (1, TRUE) or not (0, FALSE).
AUTOPILOT_ALTITUDE_SLOT_INDEX | n (FLOAT64) | Index of the slot that the autopilot will use for the altitude reference. Note that there are 3 slot indices (1, 2, 3) that you can set/get normally, however you can also target slot index 0. Writing to slot 0 will overwrite all other slots with the slot 0 value, and by default the autopilot will follow slot 0 if you have not selected any slot index.See alt_mode_slot_index for more information.
AUTOPILOT_APPROACH_ACTIVE | b (BOOL) | When TRUE (1), the autopilot is currently flying the approach Flight Plan (the last legs), and when FALSE (0) it's not.
AUTOPILOT_APPROACH_ARM | b (BOOL) | Returns TRUE (1) when the autopilot is active on the approach, once it reaches the adequate condition (in most cases, once it reaches the second-last waypoint of the flightplan).
AUTOPILOT_APPROACH_CAPTURED | b (BOOL) | Returns TRUE (1) when the lateral NAV mode is engaged and the angular deviation with the current tuned navigation frequency is less than 5°.
AUTOPILOT_APROACH_HOLD | b (BOOL) | Returns whether approach mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_ATTITUDE_HOLD | b (BOOL) | Returns whether attitude hold mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_AUTO_THROTTLE_ARM | b (BOOL) | Returns whether the autopilot auto-throttle is armed (1, TRUE) or not (0, FALSE).
AUTOPILOT_AVIONICS_MANAGED | b (BOOL) | Returns whether the autopilot has active managed avionics (1, TRUE) or not (0, FALSE).
AUTOPILOT_BACKCOURSE_HOLD | b (BOOL) | Returns whether the autopilot back course mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_BANK_HOLD_REF | n (FLOAT64) | The current bank-hold bank reference.Note that if you set this, the next frame the value will be overwritten by the engine, so you may need to write to this every game frame to ensure it maintains the required value.
AUTOPILOT_BANK_MODE | b (BOOL) | Returns whether the autopilot bank hold mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_CRUISE_SPEED | b (BOOL) | Currently not used within the simulation.
AUTOPILOT_DISENGAGED | b (BOOL) | Returns whether the autopilot has been disengaged (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_ACTIVE | b (BOOL) | Returns whether the autopilot is receiving a signal from the runway beacon and is following the slope to reach the ground (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_ARM | b (BOOL) | Returns whether the autopilot is currently active on the glideslope (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_HOLD | b (BOOL) | Returns whether the autopilot glidslope hold is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_LOCK | b (BOOL) | Returns whether the autopilot heading lock is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_LOCK_DIR | n (FLOAT64) | Specifies / Returns the locked in heading for the autopilot.
AUTOPILOT_HEADING_MANUALLY_TUNABLE | b (BOOL) | Returns whether the autopilot heading lock is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_SLOT_INDEX | n (FLOAT64) | Index of the slot that the autopilot will use for the heading reference. Note that there are 3 slot indices (1, 2, 3) that you can set/get normally, however you can also target slot index 0. Writing to slot 0 will overwrite all other slots with the slot 0 value, and by default the autopilot will follow slot 0 if you have not selected any slot index.
AUTOPILOT_MACH_HOLD | b (BOOL) | Returns whether the autopilot mach hold is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_MACH_HOLD_VAR | n (FLOAT64) | Returns the target holding mach airspeed for the autopilot.
AUTOPILOT_MANAGED_INDEX | n (FLOAT64) | Currently not used within the simulation.
AUTOPILOT_MANAGED_SPEED_IN_MACH | b (BOOL) | Returns whether the autopilot managed speed is in mach (1, TRUE) or not (0, FALSE).
AUTOPILOT_MANAGED_THROTTLE_ACTIVE | b (BOOL) | Returns whether the autopilot managed throttle is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_MAX_BANK | n (FLOAT64) | Returns the maximum banking angle for the autopilot, in radians.
AUTOPILOT_MAX_BANK_ID | n (FLOAT64) | Returns the index of the current maximum bank setting of the autopilot.
AUTOPILOT_MAX_SPEED | b (BOOL) | Currently not used within the simulation.
AUTOPILOT_NAV1_LOCK


</body>
</html>Parameters	Universal Var Member	Description
AUTOPILOT_ACTIVE	b (BOOL)	Toggles the autopilot master on/off.
AUTOPILOT_AIRSPEED_ACQUISITION	b (BOOL)	Currently not used within the simulation.
AUTOPILOT_AIRSPEED_HOLD	b (BOOL)	Returns whether airspeed hold is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_AIRSPEED_HOLD_CURRENT	b (BOOL)	Currently not used within the simulation.
AUTOPILOT_AIRSPEED_HOLD_VAR	n (FLOAT64)	Returns the target holding airspeed (in [Knot](https://docs.flightsimulator.com/html/Programming_Tools/Event_IDs/#)s) for the autopilot.
AUTOPILOT_AIRSPEED_MAX_CALCULATED	n (FLOAT64)	Can be set to limit the maximum airspeed of the autopilot to the given [kcas](https://docs.flightsimulator.com/html/Programming_Tools/Event_IDs/#) value.
AUTOPILOT_AIRSPEED_MIN_CALCULATED	n (FLOAT64)	Can be set to limit the minimum airspeed of the autopilot to the given [kcas](https://docs.flightsimulator.com/html/Programming_Tools/Event_IDs/#) value.
AUTOPILOT_ALTITUDE_ARM	b (BOOL)	Returns whether the autopilot is in altitude arm mode or not.
AUTOPILOT_ALTITUDE_LOCK	b (BOOL)	Toggles the autopilot altitude lock on or off.
AUTOPILOT_ALTITUDE_MANUALLY_TUNABLE	b (BOOL)	Returns whether the autopilot hold altitude is manually tuneable (1, TRUE) or not (0, FALSE). 
AUTOPILOT_ALTITUDE_SLOT_INDEX	n (FLOAT64)	
Index of the slot that the autopilot will use for the altitude reference. Note that there are 3 slot indices (1, 2, 3) that you can set/get normally, however you can also target slot index 0. Writing to slot 0 will overwrite all other slots with the slot 0 value, and by default the autopilot will follow slot 0 if you have not selected any slot index.

See [alt_mode_slot_index](https://docs.flightsimulator.com/html/Content_Configuration/SimObjects/Aircraft_SimO/systems_cfg.htm#alt_mode_slot_index) for more information.

AUTOPILOT_APPROACH_ACTIVE	b (BOOL)	When TRUE (1), the autopilot is currently flying the approach Flight Plan (the last legs), and when FALSE (0) it's not.
AUTOPILOT_APPROACH_ARM	b (BOOL)	Returns TRUE (1) when the autopilot is active on the approach, once it reaches the adequate condition (in most cases, once it reaches the second-last waypoint of the flightplan).
AUTOPILOT_APPROACH_CAPTURED	b (BOOL)	Returns TRUE (1) when the lateral NAV mode is engaged and the angular deviation with the current tuned navigation frequency is less than 5°.
AUTOPILOT_APROACH_HOLD	b (BOOL)	Returns whether approach mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_ATTITUDE_HOLD	b (BOOL)	Returns whether attitude hold mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_AUTO_THROTTLE_ARM	b (BOOL)	Returns whether the autopilot auto-throttle is armed (1, TRUE) or not (0, FALSE).
AUTOPILOT_AVIONICS_MANAGED	b (BOOL)	Returns whether the autopilot has active managed avionics (1, TRUE) or not (0, FALSE).
AUTOPILOT_BACKCOURSE_HOLD	b (BOOL)	Returns whether the autopilot back course mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_BANK_HOLD_REF	n (FLOAT64)	
The current bank-hold bank reference.

Note that if you set this, the next frame the value will be overwritten by the engine, so you may need to write to this every game frame to ensure it maintains the required value.

AUTOPILOT_BANK_MODE	b (BOOL)	Returns whether the autopilot bank hold mode is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_CRUISE_SPEED	b (BOOL)	Currently not used within the simulation.
AUTOPILOT_DISENGAGED	b (BOOL)	Returns whether the autopilot has been disengaged (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_ACTIVE	b (BOOL)	Returns whether the autopilot is receiving a signal from the runway beacon and is following the slope to reach the ground (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_ARM	b (BOOL)	Returns whether the autopilot is currently active on the glideslope (1, TRUE) or not (0, FALSE).
AUTOPILOT_GLIDESLOPE_HOLD	b (BOOL)	Returns whether the autopilot glidslope hold is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_LOCK	b (BOOL)	Returns whether the autopilot heading lock is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_LOCK_DIR	n (FLOAT64)	Specifies / Returns the locked in heading for the autopilot. 
AUTOPILOT_HEADING_MANUALLY_TUNABLE	b (BOOL)	Returns whether the autopilot heading lock is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_HEADING_SLOT_INDEX	n (FLOAT64)	Index of the slot that the autopilot will use for the heading reference. Note that there are 3 slot indices (1, 2, 3) that you can set/get normally, however you can also target slot index 0. Writing to slot 0 will overwrite all other slots with the slot 0 value, and by default the autopilot will follow slot 0 if you have not selected any slot index.
AUTOPILOT_MACH_HOLD	b (BOOL)	Returns whether the autopilot mach hold is enabled (1, TRUE) or not (0, FALSE).
AUTOPILOT_MACH_HOLD_VAR	n (FLOAT64)	Returns the target holding mach airspeed for the autopilot.
AUTOPILOT_MANAGED_INDEX	n (FLOAT64)	Currently not used within the simulation.
AUTOPILOT_MANAGED_SPEED_IN_MACH	b (BOOL)	Returns whether the autopilot managed speed is in mach (1, TRUE) or not (0, FALSE).
AUTOPILOT_MANAGED_THROTTLE_ACTIVE	b (BOOL)	Returns whether the autopilot managed throttle is active (1, TRUE) or not (0, FALSE).
AUTOPILOT_MAX_BANK	n (FLOAT64)	Returns the maximum banking angle for the autopilot, in radians.
AUTOPILOT_MAX_BANK_ID	n (FLOAT64)	Returns the index of the current maximum bank setting of the autopilot.
AUTOPILOT_MAX_SPEED	b (BOOL)	Currently not used within the simulation.
AUTOPILOT_NAV1_LOCK

**Reply by tbjohn** [reply]
Hi Max,
Only just seen the extensive list of variables. That should keep me busy.
Thank you again,
Trevor.
